message/validation/common_checks.go

```diff
@@ -56,25 +55,16 @@ func (mv *messageValidator) validateDutyCount(
 	msgID spectypes.MessageID,
 	msgSlot phase0.Slot,
 	validatorIndices []phase0.ValidatorIndex,
-	signerStateBySlot *treemap.Map,
+	signerStateBySlot *OperatorState,
 ) error {
-	msgEpoch := mv.netCfg.Beacon.EstimatedEpochAtSlot(msgSlot)
-	dutyCount := 0
-	signerStateBySlot.Each(func(slot any, state any) {
-		if mv.netCfg.Beacon.EstimatedEpochAtSlot(slot.(phase0.Slot)) == msgEpoch {
-			dutyCount++
-		}
-	})
-	if _, ok := signerStateBySlot.Get(msgSlot); !ok {
-		dutyCount++
-	}
+	dutyCount := signerStateBySlot.DutyCount(mv.netCfg.Beacon.EstimatedEpochAtSlot(msgSlot))
```

Comment:
is there any concurrent calls to `signerStateBySlot`? it is not protected.

Reply:
@y0sher I just added a mutex lock inside `DutyCount`